aiida/cmdline/utils/defaults.py

```diff
@@ -17,33 +17,24 @@
 from aiida.manage.configuration import get_config
 
 
-def get_default_profile(ctx, param, value):  # pylint: disable=unused-argument
+def get_default_profile():  # pylint: disable=unused-argument
     """Try to get the default profile.
 
     This should be used if the default profile should be returned lazily, at a point for example when the config
```

Comment:
I guess it is missing that it can return `None` if no default profile is set. But if you are referring to the explanation of where this should be used, that is still valid.

Reply:
I meant: this says 

> this should be used if the default profile should be returned lazily [...]Otherwise, the preference should go to calling `get_config` to load the actual config and using `config.default_profile_name`

However, now it returns the profile *name*.
I'm not sure there's really any preference for manually calling`get_config` anymore.

Reply:
I will update it to be more specific, but it does reference `config.default_profile_name`. The usage rule still applies. If one needs to know the default profile anywhere in application code, one should always go through the `config` instance. This function should only be used when this is not possible, for example when defining a default for the command line option